apps/prairielearn/python/colors.py

```diff
@@ -1,196 +1,155 @@
-    "lightgoldenrodyellow": "#FAFAD2",
-    "lightgray": "#D3D3D3",
-    "lightgrey": "#D3D3D3",
-    "lightgreen": "#90EE90",
-    "lightpink": "#FFB6C1",
-    "lightsalmon": "#FFA07A",
-    "lightseagreen": "#20B2AA",
-    "lightskyblue": "#87CEFA",
-    "lightslategray": "#778899",
-    "lightslategrey": "#778899",
-    "lightsteelblue": "#B0C4DE",
-    "lightyellow": "#FFFFE0",
-    "lime": "#00FF00",
-    "limegreen": "#32CD32",
-    "linen": "#FAF0E6",
-    "magenta": "#FF00FF",
-    "maroon": "#800000",
-    "mediumaquamarine": "#66CDAA",
-    "mediumblue": "#0000CD",
-    "mediumorchid": "#BA55D3",
-    "mediumpurple": "#9370DB",
-    "mediumseagreen": "#3CB371",
-    "mediumslateblue": "#7B68EE",
-    "mediumspringgreen": "#00FA9A",
-    "mediumturquoise": "#48D1CC",
-    "mediumvioletred": "#C71585",
-    "midnightblue": "#191970",
-    "mintcream": "#F5FFFA",
-    "mistyrose": "#FFE4E1",
-    "moccasin": "#FFE4B5",
-    "navajowhite": "#FFDEAD",
-    "navy": "#000080",
-    "oldlace": "#FDF5E6",
-    "olive": "#808000",
-    "olivedrab": "#6B8E23",
-    "orange": "#FFA500",
-    "orangered": "#FF4500",
-    "orchid": "#DA70D6",
-    "palegoldenrod": "#EEE8AA",
-    "palegreen": "#98FB98",
-    "paleturquoise": "#AFEEEE",
-    "palevioletred": "#DB7093",
-    "papayawhip": "#FFEFD5",
-    "peachpuff": "#FFDAB9",
-    "peru": "#CD853F",
-    "pink": "#FFC0CB",
-    "plum": "#DDA0DD",
-    "powderblue": "#B0E0E6",
-    "purple": "#800080",
-    "rebeccapurple": "#663399",
-    "red": "#FF0000",
-    "rosybrown": "#BC8F8F",
-    "royalblue": "#4169E1",
-    "saddlebrown": "#8B4513",
-    "salmon": "#FA8072",
-    "sandybrown": "#F4A460",
-    "seagreen": "#2E8B57",
-    "seashell": "#FFF5EE",
-    "sienna": "#A0522D",
-    "silver": "#C0C0C0",
-    "skyblue": "#87CEEB",
-    "slateblue": "#6A5ACD",
-    "slategray": "#708090",
-    "slategrey": "#708090",
-    "snow": "#FFFAFA",
-    "springgreen": "#00FF7F",
-    "steelblue": "#4682B4",
-    "tan": "#D2B48C",
-    "teal": "#008080",
-    "thistle": "#D8BFD8",
-    "tomato": "#FF6347",
-    "turquoise": "#40E0D0",
-    "violet": "#EE82EE",
-    "wheat": "#F5DEB3",
-    "white": "#FFFFFF",
-    "whitesmoke": "#F5F5F5",
-    "yellow": "#FFFF00",
-    "yellowgreen": "#9ACD32",
+"""
+Custom colors for the PrarieLearn project based on Coloraide.
+
+Based on https://gist.github.com/facelessuser/0b129c1faf7f3f59c0de40eeaaab5691/.
+"""
+import re
+from typing import Any, Dict, Optional, Tuple
+
+from coloraide import Color
+from coloraide import algebra as alg
+from coloraide.css import serialize
+from coloraide.spaces.srgb.css import sRGB
+
+# Match the pattern of a PL color name;
+# only accepts numbers, lowercase letters, and a single underscore
+RE_PL_COLORS = re.compile(r"(?i)\b([0-9a-z][0-9a-z_]{2,})\b")
+
+# Colors used in /public/stylesheets/colors.css
+# includes additional aliases (e.g, "red3" also known as "incorrect_red")
+PL_COLORS_NAME_MAP: Dict[str, Tuple[Any, ...]] = {
```

Comment:
```suggestion
PL_COLORS_NAME_MAP: dict[str, tuple[int, int, int, int]] = {
```

Reply:
I think `float, ..., float` is better here, since that's supported (even though it's not attested, yet) and the values are later divided by 255 (`values = [c / 255 for c in values]`). Thoughts?

Reply:
Ahh that makes sense, as long as floats are supported then it makes sense to annotate these as floats.